src/service/auth.service.ts

```diff
@@ -1,29 +1,32 @@
 import bcrypt from 'bcryptjs';
-import jwt from 'jsonwebtoken';
 import User from '../model/user.model';
 import {
   UserLoginCredentials,
   UserRegisterCredentials,
 } from '../type/user.type';
 import { CoreError } from '../util/error-handler';
+import generateToken from '../util/generate-token';
 
 export const register = async ({
   email,
   password,
+  confirmPassword,
   displayName,
 }: UserRegisterCredentials) => {
+  if (password !== confirmPassword)
+    throw new CoreError('Password and Confirm Password inconsistent');
+
   const existingUser = await User.findOne({ email });
 
-  if (existingUser) {
-    throw new CoreError('Email already in use');
-  }
+  if (existingUser) throw new CoreError('Email already in use');
 
   const hashedPassword = await bcrypt.hash(password, 10);
   const user = new User({ email, password: hashedPassword, displayName });
   try {
     await user.save();
+    const data = { token: generateToken(user) };
 
-    return 'Register successed';
+    return data;
   } catch {
     throw new CoreError('Register failed.');
   }
```

Comment:
don't return token when user register

Reply:
if just return success message, user should run login process

Reply:
shouldn't they be redirected to the login page after registered? That's the normally user process

Reply:
ok, just i don't like the normally user process XD,the fix is completed